### PR TITLE
refactor repo.pp - cleanup code, add case instead of if-blocks, remove old debian, only install needed repos

### PR DESCRIPTION
**Pull request by rwaffen**
- replace if-else with case
- add a "and" condition to the debian container_manager repo installation: repo shall only be installed if container_manager is crio
- add a "case" to the redhat container_manager repo installation install only the repo depending on the chosen container_manager

### CODE REVIEW

File: manifests/repo.pp [hunk 1]
```diff
@@ -2,30 +2,39 @@
...
 ) {
   case fact('os.family') {
     'Debian': {
-      if fact('os.name') == 'Debian' {
-        if Integer(fact('os.release.major')) < 10 {
-          warning('CRI-O is only available for Debian 10 and newer')
-          warning('containerD is only available for Debian 10 and newer')
+      case fact('os.name') {
+        'Debian': {
+          if Integer(fact('os.release.major')) < 10 {
```
Comment (bastelfreak) [inline manifests/repo.pp:16]: metadata.json  doesn't list Debian 9, so we could remove this if block.

Comment (rwaffen) [reply manifests/repo.pp:16]: @ananace what do you think, can we remove this, or should we keep it? i for myself don't do debian.

Comment (ananace) [reply manifests/repo.pp:16]: I think that's reasonable, Debian 9 is quite old after all

Comment (rwaffen) [reply manifests/repo.pp:16]: okay, than i will remove this part

**Review by bastelfreak — APPROVED** [review]
looks fine to me, can you please adjust the title to be more meaningful?

**Pull request merged**